android_studio_project/praca_inz_03/app/src/main/java/com/example/praca_inz_03/MainActivity.java

```diff
@@ -2,44 +2,92 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.content.DialogInterface;
 import android.content.Intent;
 import android.os.Bundle;
 import android.view.View;
 import android.widget.EditText;
 import android.util.Log;
+import android.widget.RadioButton;
+import android.widget.RadioGroup;
 import android.widget.Spinner;
+import android.widget.Toast;
 
 
 public class MainActivity extends AppCompatActivity {
     EditText pas,usr;
-    String IP="192.168.1.164";
+   //String IP="192.168.1.164";
+    String IP="192.168.0.18";
     Spinner listaSpecjalizacji;
+    RadioButton RadioPacjent, RadioLekarz;
+    RadioGroup radiogroup1;
+    String kto="";
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        radiogroup1 = findViewById(R.id.radiogroup1);
         usr = (EditText) findViewById( R.id.email);
         pas = (EditText) findViewById(R.id.haslo);
+       // RadioPacjent = findViewById(R.id.RadioPacjent);
+        //RadioLekarz = findViewById(R.id.RadioLekarz);
+        radiogroup1.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(RadioGroup group, int checkedId) {
+                switch (checkedId) {
+                    case -1:
+                        Log.d("nic", "Choices cleared!");
+                        break;
+                    case R.id.RadioPacjent:
+                        kto="Pacjent";
+
+                        break;
+                    case R.id.RadioLekarz:
+                        kto="Lekarz";
+                        break;
+                }
+            }
+        });
     }
 
     public void login(View view) {
-        String user = usr.getText().toString();
-        String pass = pas.getText().toString();
-
-        //????? tworzenie klasy bglogowanie  i tworzenie stringa z nazwa usera i haslem
-        bgLogowanie bg = new bgLogowanie(this);
-        String res=bg.execute(user,pass).toString();
-        Log.d("MainACC","Odpowiedz logowanie "+res);
+       final String user = usr.getText().toString();
+        final String pass = pas.getText().toString();
 
+        if(kto=="Pacjent") {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.